apps/total-typescript/src/pages/invoices/index.tsx

```diff
@@ -0,0 +1,89 @@
+import * as React from 'react'
+import {DocumentTextIcon} from '@heroicons/react/outline'
+import {ChevronRightIcon} from '@heroicons/react/solid'
+import {convertToSerializeForNextResponse} from '@skillrecordings/commerce-server'
+import {getSdk} from '@skillrecordings/database'
+import {GetServerSideProps} from 'next'
+import {format} from 'date-fns'
+import Layout from 'components/app/layout'
+import Link from 'next/link'
+import {getToken} from 'next-auth/jwt'
+import {Purchase} from '@skillrecordings/database'
+import {getCurrentAbility} from '@skillrecordings/ability'
+
+export const getServerSideProps: GetServerSideProps = async ({req}) => {
+  const sessionToken = await getToken({req})
+  const {getPurchasesForUser} = getSdk()
+
+  const ability = getCurrentAbility(sessionToken as any)
```

Comment:
*not this PR*: I'm imagining a thin wrapper around `getToken({ req })` that uses Zod to provide a type union where you either get `null` or you get a `token` which has a type of `{ id: string; sub: string; ... }` so that we can have better type support in the editor and two clear paths:
- if the token is `null`, redirect
- else, you have the values you need to reference the auth'd user

Reply:
👍👍